### PR TITLE
Cleanup and unit tests for `RDFSource#get_value`

### LINKED ISSUE #41: Term delegates too many methods
I re-added :send to fix #39.  @jcoyne suggested using https://github.com/rails/rails/blob/08754f12e65a9ec79633a605e986d0f1ffa4b251/activerecord/lib/active_record/querying.rb#L3-L14 and https://github.com/rails/rails/blob/master/activerecord/lib/active_record/relation/delegation.rb#L46 instead.

### PR DESCRIPTION
**Pull request by no-reply**
To make `#get_value` more consistent, this refactors `ActiveTriples::Relation` to delegate a smaller method footprint to its `#result`. This closes #41, but the fallout should be watched carefully.

I intend this as a first step toward #120.

### CODE REVIEW

File: lib/active_triples/relation.rb [hunk 1]
```diff
@@ -233,13 +254,13 @@ def class_for_property
...
+      ##
+      # @return [RDF::Term] the subject of the relation
       def rdf_subject
-        raise ArgumentError, "wrong number of arguments (#{value_arguments.length} for 1-2)" if value_arguments.length < 1 || value_arguments.length > 2
-        if value_arguments.length > 1
-          value_arguments.first
-        else
-          parent.rdf_subject
-        end
+        raise ArgumentError, 'wrong number of arguments' \
```
Comment (terrellt) [inline lib/active_triples/relation.rb]: Is there a reason for this refactor? It feels like it's harder to mentally parse the combination of ternaries, spacing, and postfix conditionals.

File: lib/active_triples/relation.rb [hunk 2]
```diff
@@ -153,8 +168,14 @@ def add_child_node(resource,object=nil)
...
 
+      ##
+      # Gives the predicate used by the Relation. Values of this object are 
+      # those that match the pattern `<rdf_subject> <predicate> [value] .`
+      #
+      # @return [RDF::Term] the predicate for this relation
       def predicate
-        property.kind_of?(RDF::URI) ? property : property_config[:predicate]
+        return property if property.is_a?(RDF::Term)
+        property_config[:predicate] unless property_config.nil?
```
Comment (terrellt) [inline lib/active_triples/relation.rb]: Can we just make property_config not nil?

File: lib/active_triples/relation.rb [hunk 3]
```diff
@@ -40,14 +45,14 @@ def clear
     def result
       parent.query(:subject => rdf_subject, :predicate => predicate)
         .each_with_object([]) do |x, collector|
-          converted_object = convert_object(x.object)
-          collector << converted_object unless converted_object.nil?
+        converted_object = convert_object(x.object)
```
Comment (terrellt) [inline lib/active_triples/relation.rb]: This spacing seems wrong.

File: lib/active_triples/relation.rb [hunk 4]
```diff
@@ -11,13 +11,18 @@ module ActiveTriples
...
   class Relation
-
+    include Enumerable
+    
     attr_accessor :parent, :value_arguments, :node_cache, :rel_args
     attr_reader :reflections
 
-    delegate *(Array.public_instance_methods - [:send, :__send__, :__id__, :class, :object_id] + [:as_json]), :to => :result
+    # delegate *(Array.public_instance_methods - [:send, :__send__, :__id__, :class, :object_id] + [:as_json]), :to => :result
+    delegate :[], :each, :empty?, :to_a, :to_ary, :<=>, :==, :equal, :===, :last,
```
Comment (terrellt) [inline lib/active_triples/relation.rb]: Is this all we need? I really don't know - does Enumerable define things like #length?

File: spec/active_triples/relation_spec.rb [hunk 5]
```diff
@@ -2,6 +2,27 @@
 require 'rdf/isomorphic'
 
 describe ActiveTriples::Relation do
+  let(:parent_resource) { double("parent resource", reflections: {}) }
+
+  subject { described_class.new(parent_resource, double("value args") ) }
+
+  describe "#predicate" do
```
Comment (terrellt) [inline spec/active_triples/relation_spec.rb]: If this needs tested, maybe it's a public method?

File: lib/active_triples/relation.rb (hunk 4 above)
Comment (no-reply) [inline lib/active_triples/relation.rb]: It defines `#length` and many things like it. http://ruby-doc.org/core-2.2.2/Enumerable.html

I suspect we'll find that this isn't all we need, but it gets us comparison, iteration, enumeration, and array style access.

My assumption is that it's an improvement.

File: lib/active_triples/relation.rb (hunk 3 above)
Comment (no-reply) [inline lib/active_triples/relation.rb]: I think rubocop barks at the four-line indent, here. I see what you mean, though. Maybe:

```ruby
statements = parent.query(:subject => rdf_subject, :predicate => predicate)
statements.each_with_object([]) do |x, collector|
  ...
end
```

File: lib/active_triples/relation.rb (hunk 2 above)
Comment (no-reply) [inline lib/active_triples/relation.rb]: yeah, I suppose. Do you see that as a blocker here if I'm engaged in a major rework of `Relation`?

File: lib/active_triples/relation.rb (hunk 3 above)
Comment (terrellt) [inline lib/active_triples/relation.rb]: :+1:

File: lib/active_triples/relation.rb (hunk 4 above)
Comment (terrellt) [inline lib/active_triples/relation.rb]: :+1:

File: lib/active_triples/relation.rb (hunk 2 above)
Comment (terrellt) [inline lib/active_triples/relation.rb]: I suppose not.

File: lib/active_triples/relation.rb (hunk 1 above)
Comment (no-reply) [inline lib/active_triples/relation.rb]: The refactor on `raise` is just to wrap the line at <= 80 chars.

The ternary operator seemed better, per: https://github.com/bbatsov/ruby-style-guide/blob/master/README.md#ternary-operator

I'm not a big fan of the long `raise` line, anyway, but don't have better ideas. Maybe just an empty line at 263?

File: spec/active_triples/relation_spec.rb (hunk 5 above)
Comment (no-reply) [inline spec/active_triples/relation_spec.rb]: Agreed. I'll make it public. That seems like useful info to have on your `Relation`.

File: lib/active_triples/relation.rb (hunk 1 above)
Comment (no-reply) [inline lib/active_triples/relation.rb]: How is that?

File: spec/active_triples/relation_spec.rb (hunk 5 above)
Comment (no-reply) [inline spec/active_triples/relation_spec.rb]: Done.

**Pull request merged**